pkg/apis/core/v1/defaults.go

```diff
@@ -125,7 +125,8 @@ func SetDefaults_Service(obj *v1.Service) {
 	// Defaults ExternalTrafficPolicy field for NodePort / LoadBalancer service
 	// to Global for consistency.
 	if (obj.Spec.Type == v1.ServiceTypeNodePort ||
-		obj.Spec.Type == v1.ServiceTypeLoadBalancer) &&
+		obj.Spec.Type == v1.ServiceTypeLoadBalancer ||
+		(obj.Spec.Type == v1.ServiceTypeClusterIP && len(obj.Spec.ExternalIPs) > 0)) &&
```

Comment:
As above (and again in other places below), you only need to check "has ExternalIPs", not "is ClusterIP and has ExternalIPs".

The comment above is now slightly incorrect. I suggest "for externally-reachable services"